docs/architecture/adr-007.md

```diff
@@ -10,88 +13,104 @@ The current way to define those events is not flexible: events are hardcoded, pa
 
 This can lead to potential issue in the long run when commands needs to be updated to add more parameters (ibc-go/ gravity bridge) or even if we need to implement more "actions". Since contracts are immutable, contract developers won't be able to have access to these functionalities after deploying their CRC20 contracts if we perform an upgrade.
 
-In this ADR, we define generic events format that can be use by smart contract developer to call any "action" upon successful contract execution. 
+In this ADR, we define generic event argument that can be extended to accept new parameters in the future. 
 
 ## Decision
 
-We define 2 specifics events that smart contract can use to trigger specific action after contract execution.
+### __CronosSendToIbc Event
 
-```
-# Trigger an action after contract execution which only involve the sender account
-__CosmosCall(string functionName, address sender,  byte[]args)
+#### Changes
 
-# Trigger an action after contract execution that require moving asset from the smart contract "account"
-__CosmosCallAfterPayment(string functionName, address payer, uint amount, byte[]args)
+- Add new argument `string channelId` to accept destination channel Id;
+- Add new argument `bytes extraData` to accept extra arguments in the future.
 
-```
+#### Signature
 
-- **functionName** corresponds to the equivalent hardcoded eventName such as `CronosSendToIBC` for the event `__CronosSendToIbc` or `CronosSendToChain` for the event `__CronosSendToChain`
-- **payer** is the address which make the payment, usually equal to `msg.sender`
-- **amount** corresponds to the `amount` that the payer has paid (either by burning or sending asset to the SC)
-- **args** corresponds to the argument in order of the event removing the `sender` and `amount` from the parameters encoded using the abi event pack
+```solidity
+// Current
+event __CronosSendToIbc(address sender, string recipient, uint256 amount);
```

Comment:
Shouldn't the recipient be of type `address` ?

Reply:
I think it's `string` because it can be a chain other than evm-compatible one?

Reply:
Yes, that's correct. The address will be an arbitrary Cosmos address